db/migrate/20191023180330_migrate_ems_ref_obj_to_ems_ref_type.rb

```diff
@@ -0,0 +1,61 @@
+class MigrateEmsRefObjToEmsRefType < ActiveRecord::Migration[5.1]
+  class EmsCluster < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+  end
+  class EmsFolder < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+  end
+  class Host < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+  end
+  class ResourcePool < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+  end
+  class Snapshot < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+  end
+  class Storage < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+  end
+  class VmOrTemplate < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    self.table_name         = "vms"
+    include ActiveRecord::IdRegions
+  end
+
+  MODELS_WITH_EMS_REF_OBJ = [EmsCluster, EmsFolder, Host, ResourcePool, Snapshot, Storage, VmOrTemplate].freeze
+
+  def up
+    MODELS_WITH_EMS_REF_OBJ.each do |klass|
+      say_with_time("Converting ems_ref_obj to ems_ref_type for #{klass.name}") do
+        klass.in_my_region.where("ems_ref_obj LIKE '%VimString%'").find_each do |obj|
+          ems_ref_obj  = YAML.load(obj.ems_ref_obj)
```

Comment:
This bit is going to require that the VimString class is deserializable forever.  Is it possible to do this migration *without* the YAML parsing?

Reply:
Definitely, I want to get the rest of the changes working then I'm going to parse the string manually so we don't need to yaml load it